### PR TITLE
Module7

### PR DESCRIPTION
**Pull request by JoLeeSoftwire**
image
my travisCI repository settings

### CODE REVIEW

**Comment by Jonesey13** [comment]
Designated Reviewer: Alex

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    - secure: "CfAXYEj34sBP3JuLguE9k6+0MXwVr924MrZ36dGMVp2Xvr2XvSyL4KEACHS9NalEl11wIOvrOmjfIj10KQI0G3SEsfDmSHPNYmkbtTQ0wrUVLFbis9PzKfCkmoDhUU0dZ0k3xMueDsHAkX6shKjJ5zxNhXwrarLgQE7TuyyBfLxnmekbkE5N8H1L38RLnF+2biyHql2SZggXwYCzRSLNQrqOWnG1XlT/N4wr/LvCxcpdeabs86AGAOUrMiISjm7oIWYrpAplPV1gtAkub0C5T1piCwaeJFcnHwXvg6pBFoSvf9Cw6mKkxrmPQ8QjH2RiZwWWJifIDSwYUCooQtYh5lqkr+HRWve6kgfbiSc99H/8i2LwV/jnJtFvrwC4jNmrt0haZH6KgJ3dTf3aOnpiCHV8gboSxZFMTJbdbc8XsNk0HYQgSysoRr0EVKs0iBqGSeqpFTHP6InroBsuYi9IDKMXGTIsJCatw0mtnMnacY2V8X8tmAoGj2bO+ueEI5hMFkdJRFx8YJoTVnv22dcHgYVghOJnF9gtOZxMaY7eUeNQFiU6ERC7LbC8TvkM+bK/VPQNyWsmyTxc2biyn22gOIXHGEyCO7I7u0v7lZew2dYVThxo73HDMyvoyxaCANHaPVV1bmgoGzX0tsJo4Y7eBo/tvkqS9a16MRoLvjDKVGk="
+
+script:
+- docker build --target test --tag todoapp-test .
+- docker run todoapp-test src/tests
+- docker run -e TRELLO_TOKEN=$TRELLO_TOKEN -e TRELLO_KEY=$TRELLO_KEY todoapp-test src/tests_e2e
```
Comment (hugh-emerson) [inline .travis.yml:12]: Shorthand for `-e TRELLO_TOKEN=$TRELLO_TOKEN` is just `-e TRELLO_TOKEN`

File: src/tests_e2e/e2e_test.py [hunk 2]
```diff
@@ -40,7 +40,12 @@ def test_app():
 
 @pytest.fixture(scope="module")
 def driver():
-    with webdriver.Chrome() as driver:
+    opts = webdriver.ChromeOptions()
+    opts.add_argument('--headless')
+    opts.add_argument('--no-sandbox')
+    opts.add_argument('--disable-dev-shm-usage')
+    with webdriver.Chrome('./chromedriver', options=opts) as driver:
+    # with webdriver.Chrome() as driver:
```
Comment (hugh-emerson) [inline src/tests_e2e/e2e_test.py:49]: If you keep something like this I'd normally also leave a comment of some sort so people don't just think it's unused commented out code.

File: src/tests_e2e/e2e_test.py [hunk 3]
```diff
@@ -40,7 +40,12 @@ def test_app():
 
 @pytest.fixture(scope="module")
 def driver():
-    with webdriver.Chrome() as driver:
+    opts = webdriver.ChromeOptions()
+    opts.add_argument('--headless')
+    opts.add_argument('--no-sandbox')
+    opts.add_argument('--disable-dev-shm-usage')
+    with webdriver.Chrome('./chromedriver', options=opts) as driver:
```
Comment (hugh-emerson) [inline src/tests_e2e/e2e_test.py:47]: Can't you keep the location of the chromedriver flexible? `webdriver.Chrome(options=opts)`?

File: README.md [hunk 4]
```diff
@@ -29,21 +29,34 @@ You should (eventually) see output similar to the following:
...
 navigate to the `src/tests` directory
 run `pytest integration_test.py`
 
 ### End to end tests
+to run them in the docker container: `docker run --env-file ./.env todoapp-test src/tests_e2e`
+
+to run them locally:
 you will need to have installed Chrome web browser, and the Selenium [chrome driver](https://chromedriver.chromium.org/downloads)
 navigate to the `src/tests_e2e` folder
+comment out lines 43-47 in `e2e_tests.py` and uncomment line 48 (don't forget to reverse this if you switch to running them in a container)
```
Comment (hugh-emerson) [inline README.md]: What's wrong with also running them in headless Chrome when running locally?

**Review by hugh-emerson — APPROVED** [review]
Looks good 👍 Some minor comments but you can merge whenever you want.

**Pull request merged**